builtin/providers/azure/resource_azure_instance.go

```diff
@@ -0,0 +1,653 @@
+package azure
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/svanharmelen/azure-sdk-for-go/management"
+	"github.com/svanharmelen/azure-sdk-for-go/management/hostedservice"
+	"github.com/svanharmelen/azure-sdk-for-go/management/osimage"
+	"github.com/svanharmelen/azure-sdk-for-go/management/virtualmachine"
+	"github.com/svanharmelen/azure-sdk-for-go/management/virtualmachineimage"
+	"github.com/svanharmelen/azure-sdk-for-go/management/vmutils"
+)
+
+const (
+	linux                = "Linux"
+	windows              = "Windows"
+	osDiskBlobStorageURL = "http://%s.blob.core.windows.net/vhds/%s.vhd"
+)
+
+func resourceAzureInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAzureInstanceCreate,
+		Read:   resourceAzureInstanceRead,
+		Update: resourceAzureInstanceUpdate,
+		Delete: resourceAzureInstanceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"image": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"size": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"subnet": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"virtual_network": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"storage": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"reverse_dns": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"automatic_updates": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"time_zone": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"username": &schema.Schema{
```

Comment:
These should not be mandatory due to chains of events like having a VM configured to join an Active Directory domain etc...